### ISSUE TITLE
linter: (jsx-a11y/autocomplete-valid): false positive for WebAuthn autofill tokens (e.g. "email webauthn")

### ISSUE THREAD

**Issue by Kramarich0** [oxc-project/oxc#26863]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "categories": {
    "correctness": "error"
  },
  "ignorePatterns": [
    "**/.next/**",
    "**/out/**",
    "**/build/**",
    "**/dist/**",
    "**/.turbo/**",
    "**/.vercel/**",
    "**/public/**",
    "**/drizzle/**",
    "**/prisma/generated/**",
    "**/prisma/migrations/**",
    "**/*.d.ts",
    "**/next-env.d.ts",
    "**/vite-env.d.ts",
    "**/routeTree.gen.ts",
    "**/messages/en.d.json.ts",
    "**/coverage/**",
    "**/report/**",
    "**/reports/**",
    "**/playwright-report/**",
    "**/test-results/**",
    "**/.stryker-tmp/**",
    "**/.trigger/**"
  ],
  "options": {
    "typeAware": true
  },
  "overrides": [
    {
      "files": ["apps/web/**/*.{ts,tsx}"],
      "rules": {
        "nextjs/no-async-client-component": "error",
        "nextjs/no-head-element": "error",
        "nextjs/no-img-element": "error",
        "nextjs/no-sync-scripts": "error"
      }
    },
    {
      "files": [
        "**/src/server/**/*.{ts,tsx,js}",
        "apps/siem-server/**/*.{ts,tsx,js}",
        "packages/shared/**/*.{ts,js}"
      ],
      "rules": {
        "nextjs/no-html-link-for-pages": "off"
      }
    },
    {
      "files": [
        "**/src/server/modules/analysis/engine/**/*.{ts,tsx}",
        "**/src/server/modules/analysis/logic/**/*.{ts,tsx}"
      ],
      "rules": {
        "typescript/no-non-null-assertion": "off"
      }
    },
    {
      "files": ["**/src/shared/ui/core/**/*.{ts,tsx}", "**/shared/ui/core/**/*.{ts,tsx}"],
      "rules": {
        "no-param-reassign": "off",
        "no-shadow": "off",
        "react/jsx-no-constructed-context-values": "off"
      }
    },
    {
      "files": [
        "packages/cli/**/*.{ts,js}",
        "scripts/**/*.{ts,js,mjs}",
        "**/scripts/**/*.{ts,js,mjs}",
        "**/prisma/seed.ts",
        "**/*.config.{ts,js,mjs}"
      ],
      "rules": {
        "no-console": "off",
        "typescript/no-floating-promises": "off",
        "typescript/no-non-null-assertion": "off",
        "typescript/no-unnecessary-condition": "off",
        "unicorn/no-process-exit": "off",
        "unicorn/no-unreadable-array-destructuring": "off"
      }
    },
    {
      "files": [
        "**/*.test.ts",
        "**/*.test.tsx",
        "**/*.spec.ts",
        "**/*.spec.tsx",
        "**/tests/**/*.{ts,tsx}"
      ],
      "rules": {
        "no-shadow": "off",
        "react/no-array-index-key": "off",
        "require-yield": "off",
        "typescript/no-non-null-assertion": "off",
        "typescript/no-unnecessary-type-assertion": "warn",
        "typescript/only-throw-error": "off",
        "typescript/unbound-method": "off",
        "vitest/expect-expect": [
          "error",
          {
            "assertFunctionNames": ["expect", "expectDenied", "expectValidationFail", "fc.assert"]
          }
        ],
        "vitest/no-focused-tests": "error",
        "vitest/prefer-expect-resolves": "error",
        "vitest/require-mock-type-parameters": "off"
      }
    }
  ],
  "plugins": [
    "eslint",
    "react",
    "react-perf",
    "unicorn",
    "typescript",
    "oxc",
    "import",
    "vitest",
    "jsx-a11y",
    "nextjs",
    "node"
  ],
  "rules": {
    "array-callback-return": ["error", { "checkForEach": false }],
    "curly": ["error", "all"],
    "eqeqeq": ["error", "always", { "null": "ignore" }],
    "import/default": "off",
    "import/namespace": "off",
    "import/no-cycle": "warn",
    "import/no-named-as-default": "off",
    "import/no-named-as-default-member": "off",
    "import/no-unassigned-import": "off",
    "no-console": ["warn", { "allow": ["warn", "error", "info"] }],
    "no-implicit-coercion": ["error", { "allow": ["!!"] }],
    "no-new": "error",
    "no-param-reassign": "error",
    "no-shadow": "off",
    "no-underscore-dangle": [
      "error",
      {
        "allow": [
          "_time",
          "_id",
          "_count",
          "_all",
          "_journal",
          "_prisma",
          "_def",
          "__dirname",
          "__filename",
          "_default",
          "_defaultClassNames",
          "_dateStr",
          "_appliedCount",
          "_exhaustiveCheck"
        ]
      }
    ],
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": "error",
    "no-unused-vars": [
      "error",
      {
        "argsIgnorePattern": "^_",
        "caughtErrorsIgnorePattern": "^_",
        "destructuredArrayIgnorePattern": "^_",
        "ignoreRestSiblings": true,
        "varsIgnorePattern": "^_"
      }
    ],
    "no-useless-concat": "error",
    "no-useless-constructor": "error",
    "no-useless-return": "error",
    "node/no-path-concat": "error",
    "oxc/approx-constant": "error",
    "oxc/misrefactored-assign-op": "error",
    "oxc/no-accumulating-spread": "error",
    "oxc/no-barrel-file": "error",
    "prefer-const": "error",
    "preserve-caught-error": "error",
    "react/iframe-missing-sandbox": "error",
    "react/jsx-key": [
      "error",
      { "checkFragmentShorthand": true, "checkKeyMustBeforeSpread": true }
    ],
    "react/jsx-no-constructed-context-values": "error",
    "react/jsx-no-script-url": "error",
    "react/jsx-no-target-blank": "error",
    "react/jsx-no-useless-fragment": "off",
    "react/no-array-index-key": "off",
    "react/no-unstable-nested-components": ["error", { "allowAsProps": true }],
    "react/react-in-jsx-scope": "off",
    "react/rules-of-hooks": "error",
    "react/self-closing-comp": "error",
    "typescript/ban-ts-comment": [
      "error",
      {
        "ts-expect-error": "allow-with-description",
        "ts-ignore": "allow-with-description"
      }
    ],
    "typescript/consistent-return": "off",
    "typescript/consistent-type-imports": [
      "error",
      {
        "fixStyle": "separate-type-imports",
        "prefer": "type-imports"
      }
    ],
    "typescript/no-base-to-string": "off",
    "typescript/no-confusing-non-null-assertion": "error",
    "typescript/no-empty-object-type": "error",
    "typescript/no-explicit-any": "warn",
    "typescript/no-extra-non-null-assertion": "error",
    "typescript/no-extraneous-class": "off",
    "typescript/no-misused-promises": [
      "error",
      {
        "checksVoidReturn": {
          "attributes": false
        }
      }
    ],
    "typescript/no-namespace": "error",
    "typescript/no-non-null-assertion": "error",
    "typescript/no-redundant-type-constituents": "off",
    "typescript/no-unnecessary-boolean-literal-compare": "error",
    "typescript/no-unnecessary-condition": [
      "error",
      {
        "allowConstantLoopConditions": "always"
      }
    ],
    "typescript/no-unnecessary-template-expression": "error",
    "typescript/no-unnecessary-type-arguments": "error",
    "typescript/no-unnecessary-type-assertion": "error",
    "typescript/no-unnecessary-type-constraint": "error",
    "typescript/no-unnecessary-type-conversion": "off",
    "typescript/no-unnecessary-type-parameters": "off",
    "typescript/no-unsafe-declaration-merging": "error",
    "typescript/no-unsafe-type-assertion": "off",
    "typescript/only-throw-error": "error",
    "typescript/prefer-includes": "error",
    "typescript/prefer-nullish-coalescing": [
      "error",
      {
        "ignorePrimitives": {
          "boolean": true,
          "string": true
        }
      }
    ],
    "typescript/prefer-optional-chain": "error",
    "typescript/restrict-template-expressions": "error",
    "typescript/return-await": ["error", "in-try-catch"],
    "typescript/switch-exhaustiveness-check": "error",
    "typescript/unbound-method": "error",
    "unicorn/catch-error-name": "error",
    "unicorn/consistent-function-scoping": "off",
    "unicorn/error-message": "error",
    "unicorn/filename-case": "error",
    "unicorn/no-abusive-eslint-disable": "error",
    "unicorn/no-accessor-recursion": "error",
    "unicorn/no-array-callback-reference": "off",
    "unicorn/no-array-fill-with-reference-type": "error",
    "unicorn/no-array-for-each": "off",
    "unicorn/no-array-reduce": "off",
    "unicorn/no-array-sort": "off",
    "unicorn/no-await-expression-member": "error",
    "unicorn/no-empty-file": "off",
    "unicorn/no-instanceof-array": "error",
    "unicorn/no-lonely-if": "off",
    "unicorn/no-null": "off",
    "unicorn/no-process-exit": "error",
    "unicorn/no-static-only-class": "error",
    "unicorn/no-unreadable-array-destructuring": "error",
    "unicorn/no-useless-promise-resolve-reject": "error",
    "unicorn/no-useless-undefined": "off",
    "unicorn/numeric-separators-style": "error",
    "unicorn/prefer-add-event-listener": "off",
    "unicorn/prefer-array-find": "error",
    "unicorn/prefer-array-flat": "error",
    "unicorn/prefer-array-flat-map": "error",
    "unicorn/prefer-array-some": "error",
    "unicorn/prefer-at": "error",
    "unicorn/prefer-date-now": "error",
    "unicorn/prefer-default-parameters": "error",
    "unicorn/prefer-dom-node-append": "error",
    "unicorn/prefer-dom-node-remove": "error",
    "unicorn/prefer-includes": "error",
    "unicorn/prefer-keyboard-event-key": "error",
    "unicorn/prefer-logical-operator-over-ternary": "off",
    "unicorn/prefer-modern-math-apis": "error",
    "unicorn/prefer-native-coercion-functions": "error",
    "unicorn/prefer-node-protocol": "error",
    "unicorn/prefer-number-properties": "error",
    "unicorn/prefer-optional-catch-binding": "error",
    "unicorn/prefer-query-selector": "off",
    "unicorn/prefer-set-has": "error",
    "unicorn/prefer-string-replace-all": "error",
    "unicorn/prefer-string-slice": "error",
    "unicorn/prefer-top-level-await": "off",
    "unicorn/require-post-message-target-origin": "error",
    "unicorn/switch-case-braces": ["error", "always"],
    "unicorn/throw-new-error": "error"
  },
  "settings": {
    "jsx-a11y": {
      "components": {
        "Accordion": "div",
        "Badge": "span",
        "Button": "button",
        "Checkbox": "input",
        "Image": "img",
        "Input": "input",
        "Label": "label",
        "Link": "a",
        "NextImage": "img",
        "Select": "select",
        "Table": "table",
        "Textarea": "textarea"
      },
      "polymorphicPropName": "asChild"
    },
    "next": {
      "rootDir": "apps/web"
    },
    "react": {
      "version": "19.0.0"
    }
  }
}

```

### What happened?
Hey! First off, thanks for this awesome library.

The jsx-a11y/autocomplete-valid rule reports a false positive when using the WebAuthn autofill token together with another valid autocomplete token.

### Repro

```tsx
export function AuthForm() {
  return (
    <input
      type="email"
      autoComplete="email webauthn"
      placeholder="user@example.com"
    />
  );
}
```

The same issue occurs with other valid combinations such as:

```tsx
<input autoComplete="username webauthn" />
<input autoComplete="current-password webauthn" />
```

### Diagnostic output

```text
`email webauthn` is not a valid value for `autocomplete`.
help: Change `email webauthn` to a valid value for `autocomplete`.
```

### Expected behavior

These values should be accepted as valid.

The WebAuthn Conditional UI specification defines webauthn as an HTML autofill token specifically intended to be combined with existing autocomplete tokens. For example, the specification gives:

```html
<input type="text" name="name" autocomplete="username webauthn">
<input type="password" name="password" autocomplete="current-password webauthn">
```

The webauthn token must be the final token in the autocomplete attribute.

### Root cause

The current implementation of `is_valid_autocomplete_value` handles two-token values only when the first token is billing or shipping:

```rust
fn is_valid_autocomplete_value(value: &str) -> bool {
    let parts: Vec<&str> = value.split_whitespace().collect();

    match parts.len() {
        1 => VALID_AUTOCOMPLETE_VALUES.contains(parts[0]),
        2 if ["billing", "shipping"].contains(&parts[0]) => {
            BILLING_AND_SHIPPING.contains(&parts[1])
        }
        _ => false,
    }
}
```

Therefore:

```text
email webauthn
username webauthn
current-password webauthn
```

all reach the `_ => false` branch, even though webauthn is a valid WebAuthn autofill token and is explicitly intended to be combined with existing autocomplete tokens.

### Suggested fix

Allow webauthn as the final token in a valid autocomplete sequence, while preserving the existing validation rules for other combinations.

For example, the two-token case could accept:

```text
<valid-autocomplete-token> webauthn
```

in addition to the existing billing/shipping combinations.

I would also suggest adding regression tests for at least:

```text
email webauthn
username webauthn
current-password webauthn
```

and ensuring that invalid combinations are still rejected.
